### ISSUE TITLE
Surface Laptop Studio - NVIDIA GPU 3050ti, not being use from the system.

### ISSUE THREAD

**Issue by Special-Niewbie** [linux-surface/linux-surface#946]
### SYSTEM

- Hardware model: `Surface Laptop Studio (Intel Core i7-11370H), NVIDIA GeForce RTX 3050 Ti, 32GB RAM , etc.`
- Kernel version: `5.9.10-surface`
- Distribution: `Ubuntu 22.04.1 LTS `
- Disk Encryption: `YES `
- Secure Boot: `YES `
- File System: `ext4/LVM`
- DKMS Installed: `(normal) DKMS`

Hi everyone, 

on my Surface Laptop Studio I have everything installed correctly, but it seems that the Nvidia Video card is not being used by the system.
The card is recognized by the system but is not used, naively I went to see the additional drivers section for the system, and by default the system was using the NVIDIA GPU drivers  `xserver-xorg-video-nouveau` . 

I did the NVIDIA driver switch (which the system gives me available) to use my Nvidia 3050ti, I tried 2/3 drivers from Nvidia list and the system looks like it doesn't use the 3050ti GPU but always uses the Intel GPU.

Even in the "**NVIDIA X server settings**" software, I don't see the settings listed of my card, below the pic:

![Schermata del 2022-09-30 20-08-15](https://user-images.githubusercontent.com/64843123/193266416-05dae1da-9e2d-4c89-8b41-f7d04814b5d7.png)

![Schermata del 2022-09-30 20-08-37](https://user-images.githubusercontent.com/64843123/193266454-fec61703-f9fb-45f2-8f22-0f8b633a1d52.png)

![Schermata del 2022-09-30 20-13-34](https://user-images.githubusercontent.com/64843123/193267186-75199f02-2c82-4d88-bf3f-64af98e95e02.png)

Where I mistake?

**Comment by qzed** [comment]
Can you try `lspci -k | grep -A 2 -i "nvidia"`? The dGPU usually isn't marked as "VGA compatible".

**Comment by Special-Niewbie** [comment]
> 

Hi @qzed , yes the resaults it's as below:

![Schermata del 2022-10-01 10-41-30](https://user-images.githubusercontent.com/64843123/193379981-c2e40371-1586-4833-8c34-6566e0d4e878.png)

So, seems the system it's using it right? 
In this case why the NVIDIA X Server setting it doesn't show anything?

**Comment by NP-chaonay** [comment]
> > 
> 
> Hi @qzed , yes the resaults it's as below:
> 
> ![Schermata del 2022-10-01 10-41-30](https://user-images.githubusercontent.com/64843123/193379981-c2e40371-1586-4833-8c34-6566e0d4e878.png)
> 
> So, seems the system it's using it right? In this case why the NVIDIA X Server setting it doesn't show anything?

what is current driver selection now? in software setting.

is it a nvidia, if yes then installing nvidia-prime and switching nvidia-prime to nvidia for testing on X11

**Comment by Special-Niewbie** [comment]
> 

Hi @NP-chaonay ,

it's really strange the situation seems that the GPU it's there , but the drivers that are you going to install in the system the GPU will not get any benefit of it and will not be used from the own system. I try to explain with the pics, which test I did till now:
- As per picture below, I have tried any driver listed available from the system, so the results are the same I just attach below one of the example:
![Schermata del 2022-10-01 15-56-01](https://user-images.githubusercontent.com/64843123/193409175-9c956293-aa9e-4f38-b5fa-728df7320e56.png)

- And the command suggested by @qzed it shows as below:
![Schermata del 2022-10-01 15-55-26](https://user-images.githubusercontent.com/64843123/193409211-d3970c63-949c-4d26-b84b-b8c76b91c959.png)

- If I check into the NVIDIA X Server Setting, for all the drivers installed and tested the resault are the same as below:
![Schermata del 2022-10-01 15-57-49](https://user-images.githubusercontent.com/64843123/193409250-90ec5a23-36f2-4ad5-ad4a-15168f9e6ca4.png)

- I have tested to install all the drivers even without Secure Boot activated (so , Secure Boot OFF from the BIOS), because I have another system with NVIDIA GPU and UBUNTU and sometimes if you have the Secure Boot activated could create some problem to integrate the Drivers into the System, but it's not this the case here (so the results it doesn't change).

- After that I have tested with the Desktop Environment X11 instead of Wayland as Default , and the result it doesn't change.

-After I check if `nvidia-prime` was installed , and it's installed, even if it doesn't give me any possibility to use it. Here it's the Pic:
![Schermata del 2022-10-01 16-03-26](https://user-images.githubusercontent.com/64843123/193409508-cbac2f3a-57a1-4ff4-8d60-c9f2d78462e5.png)

- After that, to find the way to force to switch the GPU, i have used the Terminal to do it with `sudo prime-select nvidia` and the results in the Terminal was as below:
![Schermata del 2022-10-01 16-02-10](https://user-images.githubusercontent.com/64843123/193409557-268f9cc5-dd83-4057-8eea-80a50f076de9.png)

But even with the Forced Switching from Prime, it doesn't change the result no where, the GPU seems not be used and into the `NVIDIA X Server Setting ` app it doesn't show anything as the Above Pic.

After that, because I use Conky Widgets I have one Widget that should get me any info from the Discrete GPU (NVIDIA), and this widget it takes the info into the folder address example like that " `/proc/driver/nvidia/gpus/0000:01:00.0/information` " , but into this address folder there is nothing related to the NVIDIA driver that should be downloaded and installed into it (as per another my system with NVIDIA Card). This is the Pic:
![Schermata del 2022-10-01 15-57-20](https://user-images.githubusercontent.com/64843123/193409781-415cd789-597e-4ab9-82be-e01e187d2a83.png)

Really, what can be here? Any ideas?

**Comment by qzed** [comment]
Can you make sure that you have the `nvidia-dkms` package installed? For "custom" kernels, you need the DKMS version of the driver.

**Comment by qzed** [comment]
You also might want to blacklist the nouveau driver, i.e. put
```
blacklist nouveau
```
in `/etc/modprobe.d/dgpu.conf`.

**Comment by Special-Niewbie** [comment]
> Can you make sure that you have the `nvidia-dkms` package installed? For "custom" kernels, you need the DKMS version of the driver.

Well, i didn't install it, I installed only the normal DKMS.....I feel noob now :) . Should I install NVIDIA DKMS also? 
I need to install some other DKMS?

**Comment by qzed** [comment]
Yes, install that. Also make sure that it installs correctly and doesn't print any warnings/errors. There have been some reports on the driver not compiling on 5.19.10. In that case you could try an earlier kernel version.

Whether you need other DKMS drivers depends on what external kernel drivers you need, e.g., if you want ZFS, you'll need the DKMS driver for that.

**Comment by Special-Niewbie** [comment]
> Yes, install that. Also make sure that it installs correctly and doesn't print any warnings/errors. There also has been a report on the driver not working on 5.19.10 (#936). In that case you could try an earlier kernel version.
> 
> Whether you need other DKMS drivers depends on what external kernel drivers you need, e.g., if you want ZFS, you'll need the DKMS driver for that.

YESSSSS, it works now. 
I have installed the latest `nvidia-dkms-515` and seems with the latest card  3050ti and the Kernel 5.19.10 it doesn't give me any error except the classic `missing firmware....module i915` ; so, as per another user on the topic (https://github.com/linux-surface/linux-surface/issues/936) from side everything it installed correctly seems. Here it's the result of the Terminal:

[DKMS nvidia.txt](https://github.com/linux-surface/linux-surface/files/9690395/DKMS.nvidia.txt)

Here below are the picture with the extra staff installed, and on `NVIDIA X Server Setting ` app  it appear only the Voice of PRIME (not sure if should be like that the menu or as normal should be the rest of the Menu Voices)
![Schermata del 2022-10-01 21-12-57](https://user-images.githubusercontent.com/64843123/193411490-8cd28530-b0aa-4916-83b7-ee418f75d6c9.png)

![Schermata del 2022-10-01 21-13-16](https://user-images.githubusercontent.com/64843123/193411497-852e37b4-268c-4417-8d14-21de8e07b979.png)


![Schermata del 2022-10-01 21-12-35](https://user-images.githubusercontent.com/64843123/193411504-1e444609-180f-463f-b92a-6075f0757658.png)

**Comment by qzed** [comment]
I'm unfortunately not that familiar on how to properly set up prime and make that work, but one of the easiest ways to check that the driver is working is by running `nvidia-smi`. If the driver works, that should give you some stats about the GPU.

**Comment by Special-Niewbie** [comment]
> 

I will try around, and after if i have some news I will post here and then close the topic as solved. Thank you so much!

**Comment by Special-Niewbie** [comment]
**SOLVED!!!!**

I started to test other NVIDIA DKMS drivers and I started to have the error as @qzed told me, related to the `bbswitch-dkms`. 
So, in the end I solved in this way, with PRIME by Terminal I switched to use the INTEL GPU after that I have used the `sudo apt-get purge` to delete any NVIDIA drivers and system settings installed by `apt search` function (even the Nouveau Drivers).

After that, I reboot the System, and then by Terminal I have installed again the Nouveau Drivers that was installed previously with the system (not necessary but just in case). 
From the NVIDIA list drivers of the system that gives me available, I switched from **_Nouveau_** to the normal _**515 drivers**_ and check if NVIDIA X Server Setting app take out the PRIME Option (as before start to purge all the NVIDIA drivers I cannot take out the PRIME VOICE from the menu); so and then continue with Terminal to install the **_NVIDIA DKMS 515 drivers_**. 

Then, open NVIDIA X Server Setting and with the voice PRIME appeared again , I switched to use the NVIDIA CARD (instead of INTEL), ask me to reboot the system, and on next reboot BOOM everything it's working now as the Pic below:
![Schermata del 2022-10-01 22-30-13](https://user-images.githubusercontent.com/64843123/193415214-e66b2bb5-4aa8-4462-a913-11850228e448.png)

The problems are, if you have installed previously some NVIDIA drivers and you upgrade the Kernel or to other new Drivers the system files from the previous Drivers installations could create the conflicts if you don't delete completely them.

Thank you so much @qzed  and @NP-chaonay  for the support!!!

I continue to test around and let you know with new topics if I have something else to fix...
